x/committee/proposal_handler_test.go

```diff
@@ -144,32 +144,32 @@ func (suite *ProposalHandlerTestSuite) TestProposalHandler_ChangeCommittee() {
 			suite.app = app.NewTestApp()
 			suite.keeper = suite.app.GetCommitteeKeeper()
 			suite.app = suite.app.InitializeFromGenesisStates(
-				NewCommitteeGenState(suite.app.Codec(), suite.testGenesis),
+				NewCommitteeGenState(suite.app.AppCodec(), suite.testGenesis),
 			)
-			suite.ctx = suite.app.NewContext(true, abci.Header{Height: 1, Time: testTime})
+			suite.ctx = suite.app.NewContext(true, tmproto.Header{Height: 1, Time: testTime})
 			handler := committee.NewProposalHandler(suite.keeper)
 
-			oldProposals := suite.keeper.GetProposalsByCommittee(suite.ctx, tc.proposal.NewCommittee.GetID())
+			oldProposals := suite.keeper.GetProposalsByCommittee(suite.ctx, tc.proposal.GetNewCommittee().GetID())
 
 			// Run
-			err := handler(suite.ctx, tc.proposal)
+			err := handler(suite.ctx, &tc.proposal)
 
 			// Check
 			if tc.expectPass {
 				suite.NoError(err)
 				// check committee is accurate
-				actualCom, found := suite.keeper.GetCommittee(suite.ctx, tc.proposal.NewCommittee.GetID())
+				actualCom, found := suite.keeper.GetCommittee(suite.ctx, tc.proposal.GetNewCommittee().GetID())
 				suite.True(found)
-				suite.Equal(tc.proposal.NewCommittee, actualCom)
+				testutil.AssertProtoMessageJSON(suite.T(), suite.app.AppCodec(), tc.proposal.GetNewCommittee(), actualCom)
```

Comment:
note: had to do this because it seems after proto.message marshal and unmarshals, the empty slices will get converted to nil slices, which will fail the equal assertion. So here I updated this to compare their json data instead.

Reply:
good to know, looks like go's nil and empty slices get encoded to the same thing in the protobuf wire format